examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -236,6 +277,44 @@ void RegularGridDecomposition::exchangeMigratingParticles(SharedAutoPasContainer
   }
 }
 
+void RegularGridDecomposition::reflectParticlesAtBoundaries(SharedAutoPasContainer &autoPasContainer) {
+  const std::array<double, _dimensionCount> globalBoxMin = {_globalBoxMin[0], _globalBoxMin[1], _globalBoxMin[2]};
+  const std::array<double, _dimensionCount> globalBoxMax = {_globalBoxMax[0], _globalBoxMax[1], _globalBoxMax[2]};
+  std::array<double, _dimensionCount> reflSkinMin, reflSkinMax;
+
+  for (int dimensionIndex = 0; dimensionIndex < _dimensionCount; ++dimensionIndex) {
+    // determine if boundary is reflective
+    if (_boundaryType[dimensionIndex] != autopas::BoundaryTypeOption::reflective) continue;
+    // determine if boundary
+    if (_localBoxMin[dimensionIndex] == _globalBoxMin[dimensionIndex]) {
+      reflSkinMin = globalBoxMin; reflSkinMax = globalBoxMax;
+      reflSkinMax[dimensionIndex] = _globalBoxMin[dimensionIndex] + _reflWidth;
+
+      for (auto p = autoPasContainer->getRegionIterator(reflSkinMin, reflSkinMax, autopas::IteratorBehavior::owned); p.isValid(); ++p) {
+        auto vel = p->getV();
+        // reverse velocity in dimension if towards boundary
+        if (vel[dimensionIndex] < 0) {
+          vel[dimensionIndex] *= -1;
+        }
+        p->setV(vel);
+      }
+    }
+    if (_localBoxMax[dimensionIndex] == _globalBoxMax[dimensionIndex]) {
+      reflSkinMin = globalBoxMin; reflSkinMax = globalBoxMax;
+      reflSkinMin[dimensionIndex] = _globalBoxMax[dimensionIndex] - _reflWidth;
+
+      for (auto p = autoPasContainer->getRegionIterator(reflSkinMin, reflSkinMax, autopas::IteratorBehavior::owned); p.isValid(); ++p) {
+        auto vel = p->getV();
+        // reverse velocity in dimension if towards boundary
+        if (vel[dimensionIndex] > 0) {
+          vel[dimensionIndex] *= -1;
+        }
+        p->setV(vel);
+      }
```

Comment:
this code block is 99% the same as for the lower end case. Maybe we should replace this by a (lambda) function to avoid code duplication / increase readability?

Reply:
Done. Maybe not the best implementation though, would be worth you checking